### PR TITLE
PSC uses V1 ServiceAttachment

### PR DESCRIPTION
**Pull request by swetharepakula**
PSC controller uses v1 ServiceAttachment CRD and client to process and manage service attachments

/assign @freehan @prameshj

### CODE REVIEW

**Review by freehan — APPROVED** [review]
/lgtm
/approve

File: pkg/context/context.go [hunk 1]
```diff
@@ -430,6 +431,9 @@ func (ctx *ControllerContext) generateScheme() *runtime.Scheme {
 		if err := sav1beta1.AddToScheme(controllerScheme); err != nil {
 			klog.Errorf("Failed to add ServiceAttachment CRD scheme to event recorder")
 		}
+		if err := sav1.AddToScheme(controllerScheme); err != nil {
+			klog.Errorf("Failed to add ServiceAttachment CRD scheme to event recorder")
```
Comment (prameshj) [inline pkg/context/context.go:435]: nit: log the error.. and also maybe include "V1 serviceAttachment" in the error string and "v1beta1 serviceAttachment" above.

File: pkg/e2e/adapter/psc.go [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+	legacyIng, err := crud.C.NetworkingV1beta1().ServiceAttachments(oldV1SA.Namespace).Patch(context.TODO(), oldV1SA.Name, types.MergePatchType, patchBytes, metav1.PatchOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return toPSCV1(legacyIng), nil
+}
+
+// Delete ServiceAttachment resource.
+func (crud *ServiceAttachmentCRUD) Delete(ns, name string) error {
+	new, err := crud.supportsV1API()
```
Comment (prameshj) [inline pkg/e2e/adapter/psc.go:138]: nit: "useV1" instead of "new" to be consistent?

File: pkg/e2e/adapter/psc.go [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+	}
+	klog.Warning("Using v1beta1 API")
+	return crud.C.NetworkingV1beta1().ServiceAttachments(ns).Delete(context.TODO(), name, metav1.DeleteOptions{})
+}
+
+// supportsV1API returns true if the V1 API is available
+func (crud *ServiceAttachmentCRUD) supportsV1API() (bool, error) {
+	if apiList, err := crud.C.Discovery().ServerResourcesForGroupVersion(v1PSC); err == nil {
+		for _, r := range apiList.APIResources {
+			if r.Kind == "ServiceAttachment" {
```
Comment (prameshj) [inline pkg/e2e/adapter/psc.go:154]: nit: const ServiceAttachmentKind?

**Review by prameshj — COMMENTED** [review]
looks good to me except for some minor nits.

**Pull request merged**